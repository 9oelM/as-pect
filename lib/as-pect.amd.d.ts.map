{"version":3,"file":"as-pect.amd.d.ts","sourceRoot":"","sources":["../src/util/ILogTarget.ts","../src/util/LogValue.ts","../src/util/ActualValue.ts","../src/test/TestResult.ts","../src/test/TestGroup.ts","../src/test/TestReporter.ts","../src/reporter/DefaultTestReporter.ts","../src/util/timeDifference.ts","../src/test/TestContext.ts","../src/util/IConfiguration.ts","../src/cli.ts","../src/as-pect.ts","../src/test.ts"],"names":[],"mappings":";IAAA,OAAO,EAAE,QAAQ,EAAE,sBAAmB;IAEtC,MAAM,WAAW,UAAU;QACzB,IAAI,EAAE,QAAQ,EAAE,CAAC;KAClB;;;ICJD,OAAO,EAAE,UAAU,EAAE,wBAAqB;IAE1C;;OAEG;IACH,MAAM,OAAO,QAAQ;QACnB;;;WAGG;QACH,OAAO,EAAE,MAAM,CAAK;QAEpB;;;WAGG;QACH,MAAM,EAAE,MAAM,CAAK;QAEnB;;;WAGG;QACH,KAAK,EAAE,MAAM,EAAE,CAAM;QAErB;;WAEG;QACH,OAAO,EAAE,MAAM,CAAM;QAErB;;WAEG;QACH,KAAK,EAAE,MAAM,CAAM;QAEnB;;WAEG;QACH,MAAM,EAAE,UAAU,GAAG,IAAI,CAAQ;KAClC;;;ICtCD,OAAO,EAAE,QAAQ,EAAE,sBAAmB;IAEtC;;;OAGG;IACH,MAAM,OAAO,WAAY,SAAQ,QAAQ;QACvC;;WAEG;QACH,OAAO,EAAE,OAAO,CAAS;KAC1B;;;ICXD,OAAO,EAAE,QAAQ,EAAE,sBAAyB;IAC5C,OAAO,EAAE,WAAW,EAAE,yBAA4B;IAClD,OAAO,EAAE,UAAU,EAAE,wBAA2B;IAEhD,MAAM,OAAO,UAAW,YAAW,UAAU;QAC3C;;WAEG;QACH,QAAQ,EAAE,MAAM,CAAM;QACtB;;WAEG;QACH,IAAI,EAAE,OAAO,CAAS;QACtB;;WAEG;QACH,IAAI,EAAE,MAAM,CAAK;QACjB;;WAEG;QACH,MAAM,EAAE,WAAW,GAAG,IAAI,CAAQ;QAClC;;WAEG;QACH,QAAQ,EAAE,WAAW,GAAG,IAAI,CAAQ;QACpC;;WAEG;QACH,OAAO,EAAE,MAAM,CAAM;QACrB;;WAEG;QACH,IAAI,EAAE,QAAQ,EAAE,CAAM;QACtB;;WAEG;QACH,KAAK,EAAE,MAAM,GAAG,IAAI,CAAQ;QAC5B;;WAEG;QACH,OAAO,EAAE,OAAO,CAAS;KAC1B;;;ICzCD,OAAO,EAAE,QAAQ,EAAE,sBAAyB;IAC5C,OAAO,EAAE,UAAU,EAAE,wBAA2B;IAChD,OAAO,EAAE,UAAU,EAAE,wBAAqB;IAE1C,MAAM,OAAO,SAAU,YAAW,UAAU;QACnC,gBAAgB,EAAE,MAAM,EAAE,CAAM;QAGhC,kBAAkB,EAAE,MAAM,EAAE,CAAM;QAClC,iBAAiB,EAAE,MAAM,EAAE,CAAM;QACjC,iBAAiB,EAAE,MAAM,EAAE,CAAM;QACjC,gBAAgB,EAAE,MAAM,EAAE,CAAM;QAGhC,oBAAoB,EAAE,MAAM,EAAE,CAAM;QACpC,gBAAgB,EAAE,MAAM,EAAE,CAAM;QAChC,mBAAmB,EAAE,MAAM,EAAE,CAAM;QACnC,UAAU,EAAE,OAAO,EAAE,CAAM;QAG3B,KAAK,EAAE,UAAU,EAAE,CAAM;QACzB,KAAK,EAAE,MAAM,EAAE,CAAM;QAGrB,IAAI,EAAE,QAAQ,EAAE,CAAM;QAEtB,IAAI,EAAE,MAAM,CAAM;QAClB,IAAI,EAAE,OAAO,CAAQ;QACrB,MAAM,EAAE,MAAM,CAAM;QACpB,IAAI,EAAE,MAAM,CAAK;QAEjB,IAAI,IAAI,SAAS;KASzB;;;ICxCD,OAAO,EAAE,WAAW,EAAE,yBAAsB;IAC5C,OAAO,EAAE,SAAS,EAAE,uBAAoB;IACxC,OAAO,EAAE,UAAU,EAAE,wBAAqB;IAC1C,OAAO,EAAE,QAAQ,EAAE,sBAAyB;IAG5C,MAAM,CAAC,QAAQ,OAAO,YAAY;QAChC;;;;WAIG;iBACa,OAAO,CAAC,KAAK,EAAE,WAAW,GAAG,IAAI;QAEjD;;;;WAIG;iBACa,YAAY,CAAC,KAAK,EAAE,SAAS,GAAG,IAAI;QAEpD;;;;WAIG;iBACa,aAAa,CAAC,KAAK,EAAE,SAAS,GAAG,IAAI;QAGrD;;;;;WAKG;iBACa,WAAW,CAAC,KAAK,EAAE,SAAS,EAAE,MAAM,EAAE,UAAU,GAAG,IAAI;QAEvE;;;;;WAKG;iBACa,YAAY,CAAC,KAAK,EAAE,SAAS,EAAE,MAAM,EAAE,UAAU,GAAG,IAAI;QAExE;;;;WAIG;iBACa,QAAQ,CAAC,KAAK,EAAE,WAAW,GAAG,IAAI;QAElD;;;;;WAKG;iBACa,MAAM,CAAC,KAAK,EAAE,SAAS,EAAE,IAAI,EAAE,MAAM,GAAG,IAAI;QAE5D;;;;;;WAMG;iBACa,KAAK,CAAC,QAAQ,EAAE,QAAQ,GAAG,IAAI;KAChD;;;ICnED,OAAO,EAAE,SAAS,EAAE,uBAA0B;IAC9C,OAAO,EAAE,UAAU,EAAE,wBAA2B;IAChD,OAAO,EAAE,WAAW,EAAE,yBAA4B;IAElD,OAAO,EAAE,QAAQ,EAAE,sBAAyB;IAE5C,OAAO,EAAE,YAAY,EAAE,0BAA6B;IAuEpD,MAAM,OAAO,mBAAoB,SAAQ,YAAY;QACnD,OAAO,CAAC,QAAQ,WAAW,GAAG,IAAI;QAGlC,YAAY,CAAC,OAAO,SAAS,GAAG,IAAI;QAKpC,aAAa,CAAC,OAAO,SAAS,GAAG,IAAI;QAkBrC,WAAW,CAAC,QAAQ,SAAS,EAAE,OAAO,UAAU,GAAG,IAAI;QACvD,YAAY,CAAC,QAAQ,SAAS,EAAE,MAAM,UAAU,GAAG,IAAI;QAuBvD,QAAQ,CAAC,OAAO,WAAW,GAAG,IAAI;QAqBlC,MAAM,CAAC,QAAQ,SAAS,EAAE,MAAM,MAAM,GAAG,IAAI;QAG7C,KAAK,CAAC,UAAU,QAAQ,GAAG,IAAI;KAoBhC;;;IC7KD,MAAM,CAAC,MAAM,cAAc,wCAA0E,CAAC;;;ICAtG,OAAO,EAAE,MAAM,EAAE,MAAM,2BAA2B,CAAC;IACnD,OAAO,EAAE,SAAS,EAAE,uBAAoB;IAIxC,OAAO,EAAE,YAAY,EAAE,0BAAuB;IAQ9C,MAAM,OAAO,WAAW;QAEf,IAAI,EAAE,MAAM,CAAM;QAEzB,OAAO,CAAC,UAAU,CAAkC;QAC7C,UAAU,EAAE,SAAS,EAAE,CAAM;QACpC,OAAO,CAAC,SAAS,CAAkC;QAEnD,OAAO,CAAC,IAAI,CAAuB;QAGnC,OAAO,CAAC,KAAK,CAAc;QAC3B,OAAO,CAAC,OAAO,CAAc;QAC7B,OAAO,CAAC,MAAM,CAA4B;QAC1C,OAAO,CAAC,QAAQ,CAA4B;QAErC,IAAI,EAAE,MAAM,CAAK;QACjB,IAAI,EAAE,OAAO,CAAQ;QAE5B;;WAEG;QACI,GAAG,CAAC,MAAM,MAAM,EAAE,WAAU,YAAwC,EAAE,OAAM,MAAW,GAAG,IAAI;QAgKrG;;;;;WAKG;QACI,aAAa,CAAC,GAAG,SAAS,GAAG,EAAE,GAAG,GAAG;QA0C5C;;;;;;;WAOG;QACH,OAAO,CAAC,cAAc;QAQtB;;;;WAIG;QACH,OAAO,CAAC,iBAAiB;QAUzB;;;;;WAKG;QACH,OAAO,CAAC,gBAAgB;QAKxB;;;;;;WAMG;QACH,OAAO,CAAC,eAAe;QAKvB;;;;WAIG;QACH,OAAO,CAAC,eAAe;QAKvB;;;;;;WAMG;QACH,OAAO,CAAC,cAAc;QAKtB;;;;;;;;WAQG;QACH,OAAO,CAAC,OAAO;QAaf;;;;WAIG;QACH,OAAO,CAAC,SAAS;QAcjB;;;;;WAKG;QACH,OAAO,CAAC,YAAY;QAepB;;;;WAIG;QACH,OAAO,CAAC,QAAQ;QAYhB;;WAEG;QACH,OAAO,CAAC,OAAO;QAcf;;WAEG;QACH,OAAO,CAAC,gBAAgB;QAYxB;;WAEG;QACH,OAAO,CAAC,kBAAkB;QAK1B;;WAEG;QACH,OAAO,CAAC,KAAK;QAEb;;;;;;;;WAQG;QACH,OAAO,CAAC,UAAU;QAQlB;;;;;;;WAOG;QACH,OAAO,CAAC,iBAAiB;QAQzB;;;;WAIG;QACH,OAAO,CAAC,UAAU;QAKpB;;;YAGI;QACF,OAAO,CAAC,aAAa;QAMrB;;WAEG;QACH,OAAO,CAAC,gBAAgB;QAQxB;;;;WAIG;QACH,OAAO,CAAC,kBAAkB;QAS1B;;;;WAIG;QACH,OAAO,CAAC,iBAAiB;QAQzB;;;;;WAKG;QACH,OAAO,CAAC,mBAAmB;QAS5B;;;;;WAKG;QACH,OAAO,CAAC,qBAAqB;QAW7B;;;;;;WAMG;QACH,OAAO,CAAC,uBAAuB;QAY9B;;;;WAIG;QACH,OAAO,CAAC,oBAAoB;QAS5B;;;;WAIG;QACH,OAAO,CAAC,mBAAmB;QAS3B;;;;WAIG;QACH,OAAO,CAAC,oBAAoB;QAS5B;;;;WAIG;QACH,OAAO,CAAC,kBAAkB;QAS1B;;;;;WAKG;QACH,OAAO,CAAC,oBAAoB;QAU5B;;;;;;;;;WASG;QACH,OAAO,CAAC,KAAK;KAGd;;;IC9pBD,OAAO,EAAE,YAAY,EAAE,0BAA6B;IAEpD,MAAM,WAAW,cAAc;QAC7B,CAAC,IAAI,EAAE,MAAM,GAAG,MAAM,EAAE,CAAC;KAC1B;IAED,MAAM,WAAW,cAAc;QAC7B;;WAEG;QACH,OAAO,CAAC,EAAE,MAAM,EAAE,CAAC;QACnB;;WAEG;QACH,GAAG,CAAC,EAAE,MAAM,EAAE,CAAC;QACf;;WAEG;QACH,KAAK,CAAC,EAAE,cAAc,CAAC;QACvB;;;WAGG;QACH,QAAQ,CAAC,EAAE,MAAM,EAAE,CAAC;QACpB;;WAEG;QACH,OAAO,CAAC,EAAE,GAAG,CAAC;QACd;;;WAGG;QACH,QAAQ,CAAC,EAAE,YAAY,CAAC;KACzB;;;ICdD;;;;OAIG;IACH,MAAM,UAAU,GAAG,CAAC,MAAM,MAAM,EAAE,QAyPjC;;;ICjRD,iCAAmC;IACnC,+BAAiC;IACjC,kCAAoC;IACpC,gCAAkC;IAClC,6CAA+C;IAC/C,iCAAmC;IACnC,oCAAsC;IACtC,gCAAkC;IAClC,8BAAgC;IAChC,oBAAsB"}