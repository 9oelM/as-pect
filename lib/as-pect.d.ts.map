{"version":3,"file":"as-pect.d.ts","sourceRoot":"","sources":["../src/as-pect.ts"],"names":[],"mappings":"AAAA,cAAc,oBAAoB,CAAC;AACnC,cAAc,kBAAkB,CAAC;AACjC,cAAc,qBAAqB,CAAC;AACpC,cAAc,mBAAmB,CAAC;AAClC,cAAc,gCAAgC,CAAC;AAC/C,cAAc,0BAA0B,CAAC;AACzC,cAAc,oBAAoB,CAAC;AACnC,cAAc,uBAAuB,CAAC;AACtC,cAAc,mBAAmB,CAAC;AAClC,cAAc,iBAAiB,CAAC;AAChC,cAAc,OAAO,CAAC"}