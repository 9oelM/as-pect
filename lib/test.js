"use strict";
/**
 * This file is used by the package.json "asp" script to simulate using the command line interface.
 */
require("../lib/cli/index.js").asp(process.argv.slice(2));
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoidGVzdC5qcyIsInNvdXJjZVJvb3QiOiIiLCJzb3VyY2VzIjpbIi4uL3NyYy90ZXN0LnRzIl0sIm5hbWVzIjpbXSwibWFwcGluZ3MiOiI7QUFBQTs7R0FFRztBQUNILE9BQU8sQ0FBQyxxQkFBcUIsQ0FBQyxDQUFDLEdBQUcsQ0FBQyxPQUFPLENBQUMsSUFBSSxDQUFDLEtBQUssQ0FBQyxDQUFDLENBQUMsQ0FBQyxDQUFDIn0=