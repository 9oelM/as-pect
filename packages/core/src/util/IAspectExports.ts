/**
 * This is the shape of the exported functions provided by the `as-pect` test suite.
 */
export interface IAspectExports {
  /**
   * This method calls a function pointer that matches the `() => void` type.
   *
   * @param {number} pointer - The function pointer.
   */
  __call(pointer: number): void;

  /** This method disables rtrace calls for the current test context. */
  __disableRTrace(): void;

  /**
   * This method returns the `usize[]` of the current module.
   */
  __getUsizeArrayId(): number;

  /** The exported web assembly memory. For compatibility with docs, this is explicit. */
  readonly memory: {
    readonly buffer: ArrayBuffer;
  };
}
