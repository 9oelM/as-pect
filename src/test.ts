import { asp } from "./cli";

asp(process.argv.slice(2));
