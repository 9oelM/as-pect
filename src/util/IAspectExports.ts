export interface IAspectExports {
  __call(pointer: number): void;
  __main(): void;
  __ready(): void;
  __sendActual(): void;
  __sendExpected(): void;
}
