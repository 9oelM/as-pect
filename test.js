const glob = require("glob");
const path = require("path");
const pattern = path.join(process.cwd(), "assembly/tests/**/*.ts");
console.log(pattern);

console.log(glob.sync(pattern));