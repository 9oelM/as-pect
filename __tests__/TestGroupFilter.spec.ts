import { TestContext } from "../src/test/TestContext";
import { createTestGroupFilterModule } from "./setup/createTestGroupFilterModule";
import { ASUtil } from "assemblyscript/lib/loader";
import { IAspectExports } from "../src/util/IAspectExports";

let context: TestContext;
let wasm: ASUtil & IAspectExports;

beforeEach(() => {
  return new Promise((resolve, reject) => {
    createTestGroupFilterModule({}, (err, result) => {
      if (err || !result) reject(err);
      else {
        context = result.context;
        wasm = result.wasm;
        resolve();
      }
    });
  });
});

describe("TestGroup filtering", () => {
  test("group filtering", (): void => {
    context.groupRegex = /two/i;
    context.run(wasm);
    expect(context.testGroups).toMatchSnapshot();
  });

  test("test filtering", (): void => {
    context.testRegex = /two/i;
    context.run(wasm);
    expect(context.testGroups).toMatchSnapshot();
  });

  test("test filtering with no test matching", () => {
    context.testRegex = /four/i;
    context.run(wasm);
    expect(context.testGroups).toMatchSnapshot();
  });
});
