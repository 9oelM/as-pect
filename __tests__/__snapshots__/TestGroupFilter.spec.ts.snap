// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestGroup filtering group filtering 1`] = `
Array [
  TestGroup {
    "afterAllPointers": Array [],
    "afterEachPointers": Array [],
    "beforeAllPointers": Array [],
    "beforeEachPointers": Array [],
    "describePointers": Array [],
    "logs": Array [],
    "maxSamples": Array [
      undefined,
      undefined,
      undefined,
    ],
    "maxTestRuntime": Array [
      undefined,
      undefined,
      undefined,
    ],
    "name": "group two",
    "pass": false,
    "performanceEnabled": Array [
      false,
      false,
      false,
    ],
    "reason": "Test suite group two passed successfully.",
    "reportAverage": Array [
      undefined,
      undefined,
      undefined,
    ],
    "reportMax": Array [
      undefined,
      undefined,
      undefined,
    ],
    "reportMedian": Array [
      undefined,
      undefined,
      undefined,
    ],
    "reportMin": Array [
      undefined,
      undefined,
      undefined,
    ],
    "reportStandardDeviation": Array [
      undefined,
      undefined,
      undefined,
    ],
    "reportVariance": Array [
      undefined,
      undefined,
      undefined,
    ],
    "roundDecimalPlaces": Array [
      undefined,
      undefined,
      undefined,
    ],
    "testFunctionPointers": Array [
      7,
      8,
      9,
    ],
    "testMessages": Array [
      "",
      "",
      "",
    ],
    "testNames": Array [
      "test one",
      "test two",
      "test three",
    ],
    "testThrows": Array [
      false,
      false,
      true,
    ],
    "tests": Array [
      TestResult {
        "actual": null,
        "average": 0,
        "decimalPlaces": 3,
        "end": 5831.330077886581,
        "expected": null,
        "hasAverage": false,
        "hasMax": false,
        "hasMedian": false,
        "hasMin": false,
        "hasStdDev": false,
        "hasVariance": false,
        "logs": Array [],
        "max": 0,
        "median": 0,
        "message": "",
        "min": 0,
        "name": "test one",
        "negated": false,
        "pass": true,
        "performance": false,
        "rawVariance": 0,
        "runTime": -0.7732858657836914,
        "stack": null,
        "start": 5830.556792020798,
        "stdDev": 0,
        "times": Array [
          0.115,
        ],
        "variance": 0,
      },
      TestResult {
        "actual": null,
        "average": 0,
        "decimalPlaces": 3,
        "end": 5897.369734048843,
        "expected": null,
        "hasAverage": false,
        "hasMax": false,
        "hasMedian": false,
        "hasMin": false,
        "hasStdDev": false,
        "hasVariance": false,
        "logs": Array [],
        "max": 0,
        "median": 0,
        "message": "",
        "min": 0,
        "name": "test two",
        "negated": false,
        "pass": false,
        "performance": false,
        "rawVariance": 0,
        "runTime": -65.90146112442017,
        "stack": "Error: not a string: 0
    at start:assembly/jest-filter~anonymous|1~anonymous|1 (wasm-function[41]:15)
    at assembly/internal/call/__call (wasm-function[50]:7)",
        "start": 5831.468272924423,
        "stdDev": 0,
        "times": Array [
          65.843,
        ],
        "variance": 0,
      },
      TestResult {
        "actual": null,
        "average": 0,
        "decimalPlaces": 3,
        "end": 5897.404093980789,
        "expected": null,
        "hasAverage": false,
        "hasMax": false,
        "hasMedian": false,
        "hasMin": false,
        "hasStdDev": false,
        "hasVariance": false,
        "logs": Array [],
        "max": 0,
        "median": 0,
        "message": "",
        "min": 0,
        "name": "test three",
        "negated": true,
        "pass": false,
        "performance": false,
        "rawVariance": 0,
        "runTime": -0.02001190185546875,
        "stack": null,
        "start": 5897.384082078934,
        "stdDev": 0,
        "times": Array [
          0.004,
        ],
        "variance": 0,
      },
    ],
    "time": 68.147,
    "todoPointers": Array [],
    "todos": Array [],
    "willRun": true,
  },
]
`;

exports[`TestGroup filtering test filtering 1`] = `
Array [
  TestGroup {
    "afterAllPointers": Array [],
    "afterEachPointers": Array [],
    "beforeAllPointers": Array [],
    "beforeEachPointers": Array [],
    "describePointers": Array [],
    "logs": Array [],
    "maxSamples": Array [
      undefined,
    ],
    "maxTestRuntime": Array [
      undefined,
    ],
    "name": "group three",
    "pass": false,
    "performanceEnabled": Array [
      false,
    ],
    "reason": "Test suite group three passed successfully.",
    "reportAverage": Array [
      undefined,
    ],
    "reportMax": Array [
      undefined,
    ],
    "reportMedian": Array [
      undefined,
    ],
    "reportMin": Array [
      undefined,
    ],
    "reportStandardDeviation": Array [
      undefined,
    ],
    "reportVariance": Array [
      undefined,
    ],
    "roundDecimalPlaces": Array [
      undefined,
    ],
    "testFunctionPointers": Array [
      12,
    ],
    "testMessages": Array [
      "",
    ],
    "testNames": Array [
      "test two",
    ],
    "testThrows": Array [
      false,
    ],
    "tests": Array [
      TestResult {
        "actual": null,
        "average": 0,
        "decimalPlaces": 3,
        "end": 6564.01185798645,
        "expected": null,
        "hasAverage": false,
        "hasMax": false,
        "hasMedian": false,
        "hasMin": false,
        "hasStdDev": false,
        "hasVariance": false,
        "logs": Array [],
        "max": 0,
        "median": 0,
        "message": "",
        "min": 0,
        "name": "test two",
        "negated": false,
        "pass": false,
        "performance": false,
        "rawVariance": 0,
        "runTime": -0.5875160694122314,
        "stack": "Error: not a string: 0
    at start:assembly/jest-filter~anonymous|2~anonymous|1 (wasm-function[45]:15)
    at assembly/internal/call/__call (wasm-function[50]:7)",
        "start": 6563.424341917038,
        "stdDev": 0,
        "times": Array [
          0.539,
        ],
        "variance": 0,
      },
    ],
    "time": 0.635,
    "todoPointers": Array [],
    "todos": Array [],
    "willRun": true,
  },
  TestGroup {
    "afterAllPointers": Array [],
    "afterEachPointers": Array [],
    "beforeAllPointers": Array [],
    "beforeEachPointers": Array [],
    "describePointers": Array [],
    "logs": Array [],
    "maxSamples": Array [
      undefined,
    ],
    "maxTestRuntime": Array [
      undefined,
    ],
    "name": "group two",
    "pass": false,
    "performanceEnabled": Array [
      false,
    ],
    "reason": "Test suite group two passed successfully.",
    "reportAverage": Array [
      undefined,
    ],
    "reportMax": Array [
      undefined,
    ],
    "reportMedian": Array [
      undefined,
    ],
    "reportMin": Array [
      undefined,
    ],
    "reportStandardDeviation": Array [
      undefined,
    ],
    "reportVariance": Array [
      undefined,
    ],
    "roundDecimalPlaces": Array [
      undefined,
    ],
    "testFunctionPointers": Array [
      8,
    ],
    "testMessages": Array [
      "",
    ],
    "testNames": Array [
      "test two",
    ],
    "testThrows": Array [
      false,
    ],
    "tests": Array [
      TestResult {
        "actual": null,
        "average": 0,
        "decimalPlaces": 3,
        "end": 6564.496294021606,
        "expected": null,
        "hasAverage": false,
        "hasMax": false,
        "hasMedian": false,
        "hasMin": false,
        "hasStdDev": false,
        "hasVariance": false,
        "logs": Array [],
        "max": 0,
        "median": 0,
        "message": "",
        "min": 0,
        "name": "test two",
        "negated": false,
        "pass": false,
        "performance": false,
        "rawVariance": 0,
        "runTime": -0.4530971050262451,
        "stack": "Error: not a string: 0
    at start:assembly/jest-filter~anonymous|1~anonymous|1 (wasm-function[41]:15)
    at assembly/internal/call/__call (wasm-function[50]:7)",
        "start": 6564.04319691658,
        "stdDev": 0,
        "times": Array [
          0.437,
        ],
        "variance": 0,
      },
    ],
    "time": 0.469,
    "todoPointers": Array [],
    "todos": Array [],
    "willRun": true,
  },
  TestGroup {
    "afterAllPointers": Array [],
    "afterEachPointers": Array [],
    "beforeAllPointers": Array [],
    "beforeEachPointers": Array [],
    "describePointers": Array [],
    "logs": Array [],
    "maxSamples": Array [
      undefined,
      undefined,
    ],
    "maxTestRuntime": Array [
      undefined,
      undefined,
    ],
    "name": "group one",
    "pass": false,
    "performanceEnabled": Array [
      false,
      false,
    ],
    "reason": "Test suite group one passed successfully.",
    "reportAverage": Array [
      undefined,
      undefined,
    ],
    "reportMax": Array [
      undefined,
      undefined,
    ],
    "reportMedian": Array [
      undefined,
      undefined,
    ],
    "reportMin": Array [
      undefined,
      undefined,
    ],
    "reportStandardDeviation": Array [
      undefined,
      undefined,
    ],
    "reportVariance": Array [
      undefined,
      undefined,
    ],
    "roundDecimalPlaces": Array [
      undefined,
      undefined,
    ],
    "testFunctionPointers": Array [
      2,
      4,
    ],
    "testMessages": Array [
      "",
      "",
    ],
    "testNames": Array [
      "test two",
      "test two",
    ],
    "testThrows": Array [
      false,
      true,
    ],
    "tests": Array [
      TestResult {
        "actual": null,
        "average": 0,
        "decimalPlaces": 3,
        "end": 6564.998476982117,
        "expected": null,
        "hasAverage": false,
        "hasMax": false,
        "hasMedian": false,
        "hasMin": false,
        "hasStdDev": false,
        "hasVariance": false,
        "logs": Array [],
        "max": 0,
        "median": 0,
        "message": "",
        "min": 0,
        "name": "test two",
        "negated": false,
        "pass": false,
        "performance": false,
        "rawVariance": 0,
        "runTime": -0.4757518768310547,
        "stack": "Error: not a string: 0
    at start:assembly/jest-filter~anonymous|0~anonymous|1 (wasm-function[33]:15)
    at assembly/internal/call/__call (wasm-function[50]:7)",
        "start": 6564.522725105286,
        "stdDev": 0,
        "times": Array [
          0.454,
        ],
        "variance": 0,
      },
      TestResult {
        "actual": null,
        "average": 0,
        "decimalPlaces": 3,
        "end": 6565.500658988953,
        "expected": null,
        "hasAverage": false,
        "hasMax": false,
        "hasMedian": false,
        "hasMin": false,
        "hasStdDev": false,
        "hasVariance": false,
        "logs": Array [],
        "max": 0,
        "median": 0,
        "message": "",
        "min": 0,
        "name": "test two",
        "negated": true,
        "pass": true,
        "performance": false,
        "rawVariance": 0,
        "runTime": -0.4863240718841553,
        "stack": null,
        "start": 6565.0143349170685,
        "stdDev": 0,
        "times": Array [
          0.469,
        ],
        "variance": 0,
      },
    ],
    "time": 0.998,
    "todoPointers": Array [],
    "todos": Array [],
    "willRun": true,
  },
]
`;

exports[`TestGroup filtering test filtering with no test matching 1`] = `Array []`;
